apps/hash-frontend/src/pages/[shortname]/entities/[entity-uuid].page/entity-editor/properties-section/property-table/cells/value-cell/inputs/number-or-text-input.tsx

```diff
@@ -1,35 +1,95 @@
-import { TextField } from "@hashintel/design-system";
+import { TextField, TextFieldProps } from "@hashintel/design-system";
+import { DataTypeWithMetadata } from "@local/hash-subgraph";
 
 import { CellInputProps } from "./types";
 
 export const NumberOrTextInput = ({
+  expectedType,
+  onBlur,
   onChange,
+  onEnterPressed,
   value,
   isNumber,
-  onEnterPressed,
-}: CellInputProps<number | string> & {
+}: CellInputProps<number | string | undefined> & {
+  onBlur?: TextFieldProps["onBlur"];
+  expectedType: DataTypeWithMetadata["schema"];
   isNumber: boolean;
   onEnterPressed?: () => void;
 }) => {
+  const minLength =
+    "minLength" in expectedType ? expectedType.minLength : undefined;
+  const maxLength =
+    "maxLength" in expectedType ? expectedType.maxLength : undefined;
+  const minimum =
+    "minimum" in expectedType
+      ? expectedType.minimum
+      : "exclusiveMinimum" in expectedType &&
+        typeof expectedType.exclusiveMinimum === "number"
+      ? expectedType.exclusiveMinimum + 1
+      : undefined;
+  const maximum =
+    "maximum" in expectedType
+      ? expectedType.minimum
+      : "exclusiveMaximum" in expectedType &&
+        typeof expectedType.exclusiveMaximum === "number"
+      ? expectedType.exclusiveMaximum - 1
+      : undefined;
+  const step =
+    "multipleOf" in expectedType ? expectedType.multipleOf : undefined;
+
+  const format = "format" in expectedType ? expectedType.format : undefined;
+
+  let inputType: TextFieldProps["type"] = isNumber ? "number" : "text";
+  switch (format) {
+    case "date-time":
+      inputType = "datetime-local";
```

Comment:
Will this use the local timezone?

Reply:
It will use no timezone – there will be no timezone added to the end. This is the only option for native HTML form. If we want a timezone we have to either guess from the user's browser, or ask the user to specify one to store it with